vent/sqlsol/parser_test.go

```
@@ -54,7 +55,144 @@ func TestParseEvents(t *testing.T) {
 		require.Equal(t, false, col.Primary)
 	})
 
-	t.Run("returns an error if the json file has an invalid format", func(t *testing.T) {
-		// TODO
+	t.Run("returns an error if the json is malformed", func(t *testing.T) {
+
+		badJSON := `[
+			{
+				"TableName" : "UserAccounts",
+				"Event"  : {
+					"anonymous": false,
+					"inputs": [{
+						"indexed": false,
+						"name": "userName",
+						"type": "string"
+					}, {
+						"indexed": false,
+						"name": "userAddress",
+						"type": "address"
+					}, {
+						"indexed": false,
+						"name": "UnimportantInfo",
+						"type": "uint"
+					}],
+					"name": "UpdateUserAccount",
+				},
+				"Columns"  : {
+					"userAddress" : {"name" : "address", "primary" : true},
+					"userName": {"name" : "username", "primary" : false}
+		]`
+
+		byteValue := []byte(badJSON)
+		_, err := sqlsol.ParseEventConfig(byteValue)
+		require.Error(t, err)
 	})
+
+	t.Run("returns an error if the json file does not exists", func(t *testing.T) {
+		filePath := fmt.Sprintf("%s/src/github.com/hiturria/bosmarmot/vent/sqlsol/test/sqlsol_example_notexists.json", os.Getenv("GOPATH"))
+
+		_, err := sqlsol.ReadFile(filePath)
+		require.Error(t, err)
+	})
+
+	t.Run("returns an error if needed json fields are missing", func(t *testing.T) {
```

Is this a test that should expect `sqlsol.ParseEventConfig` to return an error? Asking that because it says that in the name of the test, but the call to the `sqlsol.ParseEventConfig` method is not returning an error.

---

which one?
one of them only checks for the presence of the file (so ParseEventConfig is not needed), in fact it really tests ReadFile function. 
In the other case the json is malformed and ParseEventConfig should return an error

---

I'm asking just for this test that starts on line 97 :)

---

you're right, it should return an error cause there are missing fields...